src/main/kotlin/tanvd/grazi/spellcheck/GraziSpellchecker.kt

```diff
@@ -96,4 +102,30 @@ object GraziSpellchecker : GraziLifecycle {
     override fun reset() = init()
 
     override fun reInit() = init()
+
+    override fun update(prevState: GraziConfig.State?, newState: GraziConfig.State, project: Project) {
+        if (prevState?.enabledSpellcheck == newState.enabledSpellcheck) return
+
+        if (newState.enabledSpellcheck) {
+            if (!ApplicationManager.getApplication().isUnitTestMode) {
```

Comment:
Better simplify all conditions and add unitTest check in the 107 line with return